pandas/core/groupby/ops.py

```diff
@@ -625,7 +625,15 @@ def _aggregate_series_pure_python(self, obj, func):
         splitter = get_splitter(obj, group_index, ngroups, axis=self.axis)
 
         for label, group in splitter:
-            res = func(group)
+            try:
+                res = func(group)
```

Comment:
Do you know why this throws in the first place? Haven't deeply examined the entire flow but wondering if the underlying function call should account for this rather than catching here.

Reply:
The underlying problem is that pd.DataFrame.sum(pd.Series(), axis=1) raises a TypeError in Py2, but a ValueError in Py3.